### ISSUE TITLE
mirror matrix invites to irc

### ISSUE THREAD

**Issue by ashkitten** [matrix-org/matrix-appservice-irc#1504]
**Is your feature request related to a problem? Please describe.**
the only way to invite someone to an invite-only room is to either create an invite exception or INVITE them on irc-side, otherwise they will join on the matrix side but be unable to join on the irc side. this breaks message sending. currently to do it properly from matrix you need to use the unintuitive `!cmd INVITE <irc username> <channel>`

**Describe the solution you'd like**
invites from matrix should be mirrored to irc if possible.

**Describe alternatives you've considered**
if invites are not mirrored, the appservice bot should set the permissions on the bridge room so that it is the only one allowed to invite. this at least reflects the reality that inviting from matrix doesn't work properly.

**Comment by Mikaela** [comment]
I am not sure what exactly you are trying to do and where, but:

* I think new rooms default that only users with PL50 are able to invite
* on IRC usually only ops (mode +o) which match Matrix PL50 (moderator) are able to invite
  * unless there is a mode like Charybdis/Solanum `+g` set in which case everyone can invite.

**Comment by ashkitten** [comment]
yes, but if the irc channel is invite only and a matrix moderator invites a matrix user to the room, they won't be able to join the irc side anyway because the invite is not mirrored; the user joins the matrix room but not the irc channel.

**Comment by Mikaela** [comment]
I just reproduced this on libera.chat with `BridgeVersion: 0.31.0`, so you are correct.

I think this used to work with freenode so it looks like a regression.

**What I did:**

1. op a Matrix user on libera.chat
2. leave channel with IRC client
3. `/invite @ircuser:libera.chat`
4. `@ircuser:libera.chat` joins room and leaves instantly with a message `Group chat not supported.`
5. IRC user sees no notification on the invite.
